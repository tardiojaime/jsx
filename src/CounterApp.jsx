import React, { useState } from "react";
import PropTypes from "prop-types";
export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(11);

    const handle = () => {
        setCounter(counter + 1);
    }

  return (
    <>
      <h1>Counter App {value}</h1>
      <h2>{counter}</h2>
      <button onClick={handle}>incrementar</button>
    </>
  );
};
CounterApp.propTypes = {
  value: PropTypes.number,
};
CounterApp.defaultProps = {
  value: 21,
};
