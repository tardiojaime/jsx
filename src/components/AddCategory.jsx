import { useState } from "react"
import PropTypes from "prop-types"
export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) =>{
        setInputValue(event.target.value)
    }
    const onSumbit = (event)=>{
        event.preventDefault();
        if(inputValue.trim().length <= 2) return;
        onNewCategory(inputValue.trim())
        setInputValue('');
    }
  return (
    <form onSubmit={ (event) => onSumbit(event)} aria-label="form" >
    {/* <input onChange={(event)=>onInputChange(event)} type="text" placeholder='Buscar gifs' value={inputValue}/> */}
    <input onChange={onInputChange} type="text" placeholder='Buscar gifs' value={inputValue}/>
    </form>
  )
}
AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}