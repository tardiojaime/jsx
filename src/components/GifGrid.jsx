
import PropTypes from 'prop-types'
import { getGifs } from '../helpers/getGifs'
import { useEffect, useState } from 'react'
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

    const {images, isLoading} = useFetchGifs(category);
    return (
        <>
            <h3>{category}</h3>
            {
                isLoading && ( <h4>Cargando...</h4> )
            }
            <div className="card-grid">
                {images.map(img => (
                    <GifItem key={img.id} {...img} />
                ))}
            </div>
        </>
    )
}
GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}