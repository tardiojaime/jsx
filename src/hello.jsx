/* 
se recomienda seguir la notacion pascal case
    ** camel case -> contarElementos
    ** pascal case -> ContarElementos
en los nombres de los componentes
 */

export function Hello() {
  return (
    <>
      <h1>hola mundoss</h1>
    </>
  );
}


/* si se exporta de esta forma
la importacion debe ser de la siguiente forma
import componente 'ruta/hello.js'
 ### el componente puede tomar cualquier nombre
 puesto que a de tomar el componente esportado
export default Hello; */

/* 
export function App(){

} 
con este tipo de exportacion la importacion 
es de la siguiente manera
exportacion independiente
import  {Hello}  from './hello'
*/