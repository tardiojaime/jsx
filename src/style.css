h1{
    color: #590a9b;
}