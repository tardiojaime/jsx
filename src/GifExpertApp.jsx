import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

function GifExpertApp() {

  const [categories, setCategories] = useState(['Dragon Ball']);
  
  const onAddCategory = (newCategory) =>{
    if(categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories])
    //...Desustructuracion
    //setCategories((cat)=>[...cat, 'Pokemon']);
  }
  return (
    <>
    {/* title */}
      <h1>Gif Expert App</h1>
    {/* input */}
    <AddCategory 
    /* setCategories = { setCategories } */
    onNewCategory= { event =>onAddCategory(event)}
    />
    {/* listado de Gif */}
    <ol>
      { categories.map(cate =>(
        <GifGrid key={cate} category={ cate }/>
      ))}
    </ol>
    </>
  )
}

export default GifExpertApp
