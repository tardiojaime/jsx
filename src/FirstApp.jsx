import React from 'react'
import './style.css';
import PropTypes, { number, string } from 'prop-types'
/* 
para no utilizar una etiqueta padre en el componente 
podemos utilizar Fragment de react, una forma mas rapida
es utilizar <></> 
puesto que un componente solo puede retornar un componente
*/
/* 
se recomienda utilizar variable fuera del componente variables 
que no necesitan ser procesados
react no puede imprimir objetos
012 a continuacion
*/
/* 
export const FirstApp = (props) => {
desustructuracion de props  
export const FirstApp = ({title}) => {
*/

export const FirstApp = ({ title, numero }) => {
  return (
    <>
    <h1> { title +' '+ numero } </h1>
    <p>Tardio Jaime</p>
    </>
  )
}
FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  numero: PropTypes.number
}
FirstApp.defaultProps = {
  title: 'no hay titulo'
}
/* 
PropTypes, defenir los props
 */