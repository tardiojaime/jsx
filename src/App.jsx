import { useState } from 'react'
import './App.css'
import { FirstApp } from './FirstApp'
function App() {
  const [count, setCount] = useState(0)

  return (
    <>
      <h1>Vite + React</h1>
    <FirstApp/>
    </>
  )
}

export default App
