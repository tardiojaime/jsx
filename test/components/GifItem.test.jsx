import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components"

describe('Testing al componente <GitItem>', () => { 

    const title = 'saitama'
    const url = 'https://one-puch.com/saitama.png'
    test('Prueba de snapshot', () => { 
        const {container } = render(<GifItem title={ title } url= { url }/>)
        expect(container).toMatchSnapshot()
     })

     test('debe mostrar la imagen', () => { 
        render(<GifItem title={ title } url= { url}/>)
        const {src, alt }= screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
      })

 })