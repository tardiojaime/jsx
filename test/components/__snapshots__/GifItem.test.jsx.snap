// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing al componente <GitItem> Prueba de snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://one-puch.com/saitama.png"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
